docs/src/main/tut/diet.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: default
+title:  "Diet"
+source: "core/src/main/scala/Diet.scala"
+---
+# Diet
+
+`Diet` is a Discrete Interval Encoding Tree. It stores subset of types that have a total order. The types are also required to have a successor and predecessor generator. 
+
+The discrete interval encoding tree is based on the observation that the set of integers { i| a<=i<=b } can be perfectly represented by the closed interval [a, b].
+
+`Diet` is a binary search tree where each node contains a set of continuos values. If a value is inserted and it fills the hole between to nodes (sets) then the two set become one since all value in the new set are consecutive.
+
+## Best and Worst Case Analysis.
+
+The best case is when there is no holes in the stored set, so the tree contains only one node (a single interval). In this case, it has ***O***(1) space requirement, inserting, deleting and finding a value is in constant time. 
+
+In the worst case scenario, there is a hole of size one (1) between each node and each node is in fact a set on size one (1). Under these circumstances, it behaves as a regular binary search tree. 
```

Comment:
if its not balanced, it is a linked list in the worst case, right? if you keep adding things in increasing order they keep getting added to the right most leaf